### PR TITLE
WIP: Convert to TypeScript

### PR DESCRIPTION
**Pull request by buggy**
*Issue #, if available:*

#1028

*Description of changes:*

Renames the .js/.jsx files to .ts/.tsx

The last PR to complete the TypeScript conversion became too complex. I'll submit it as 2 PR's. This one will rename the files and a lot of @ts-ignore or any. The next will replace those with proper interfaces.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by buggy** [comment]
@powerful23 Can you see why this is failing to build on CircleCI? If I follow the steps from the local development wiki page then everything builds and tests pass.

**Comment by powerful23** [comment]
@buggy the build fails because:
```
3 import API from '@aws-amplify/api';
                  ~~~~~~~~~~~~~~~~~~


src/API/GraphQL/Connect.ts:4:31 - error TS2307: Cannot find module '@aws-amplify/api/lib/types'.

4 import { GraphQLResult } from '@aws-amplify/api/lib/types';
                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~


src/Amplify-UI/Amplify-UI-Components-React.tsx:3:20 - error TS2307: Cannot find module '@aws-amplify/core'.

3 import { JS } from '@aws-amplify/core';
                     ~~~~~~~~~~~~~~~~~~~


src/Amplify-UI/Amplify-UI-Components-React.tsx:6:28 - error TS2307: Cannot find module '@aws-amplify/ui'.

6 import * as AmplifyUI from '@aws-amplify/ui';
                             ~~~~~~~~~~~~~~~~~


src/AmplifyMessageMap.ts:14:22 - error TS2307: Cannot find module '@aws-amplify/core'.

14 import { I18n } from '@aws-amplify/core';
                        ~~~~~~~~~~~~~~~~~~~


src/AmplifyUI.tsx:16:20 - error TS2307: Cannot find module '@aws-amplify/core'.

16 import { JS } from '@aws-amplify/core';
                      ~~~~~~~~~~~~~~~~~~~


src/Analytics/trackLifecycle.tsx:16:23 - error TS2307: Cannot find module '@aws-amplify/analytics'.

16 import Analytics from '@aws-amplify/analytics';
                         ~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by buggy** [comment]
@powerful23 I can see the error from the circleci output. What I don't understand is why this error is occurring during the circleci build. If I follow the local development instructions from the wiki there is no error. I can only assume there is a difference between the circleci build process and the local development instructions but I don't know what.

**Comment by powerful23** [comment]
@buggy I am not sure. In theory the circle ci is just running the scripts in the `package.json`. Btw, I tried to fork your pull request and run build in my local but got those errors:
```
src/Storage/S3Album.tsx:123:14 - error TS2339: Property 'setState' does not exist on type 'S3Album'.

123         this.setState({ theme: Object.assign({}, theme) });
                 ~~~~~~~~


src/Storage/S3Album.tsx:131:18 - error TS2339: Property 'props' does not exist on type 'S3Album'.

131         if (this.props.path === prevProps.path &&
                     ~~~~~


src/Storage/S3Album.tsx:132:18 - error TS2339: Property 'props' does not exist on type 'S3Album'.

132             this.props.ts === prevProps.ts &&
                     ~~~~~


src/Storage/S3Album.tsx:133:18 - error TS2339: Property 'props' does not exist on type 'S3Album'.

133             this.props.select === prevProps.select
                     ~~~~~
```

**Comment by buggy** [comment]
@powerful23 That's really odd. Can you confirm that line 23 of `src/Storage/S3Album.tsx` in your fork contains `Component<any, any>` and not `Component`?

File: packages/aws-amplify-react/src/Auth/Provider/withGoogle.tsx [hunk 1]
```diff
@@ -24,11 +24,16 @@ import {
     SignInButtonContent
 } from '../../Amplify-UI/Amplify-UI-Components-React';
 import Constants from '../common/constants';
+import { any } from 'bluebird';
```
Comment (eknowles) [inline packages/aws-amplify-react/src/Auth/Provider/withGoogle.tsx]: shadow var here, should import this `as`

File: packages/aws-amplify-react/package.json [hunk 2]
```diff
@@ -61,10 +66,16 @@
       "/node_modules/"
     ],
     "moduleNameMapper": {
-      "\\.(css|less|scss|sass)$": "<rootDir>/__mocks__/styleMock.js"
+      "\\.(css|less|scss|sass)$": "<rootDir>/__mocks__/styleMock.ts"
     }
   },
   "dependencies": {
+    "@aws-amplify/analytics": "1.2.8-unstable.2",
```
Comment (powerful23) [inline packages/aws-amplify-react/package.json]: We should not add those as dependencies. Is this the reason why the build process failed?

Comment (buggy) [reply packages/aws-amplify-react/package.json]: Yes. That's why the build process was failing.

File: packages/aws-amplify-react/src/Auth/Provider/withGoogle.tsx (hunk 1 above)
Comment (buggy) [reply packages/aws-amplify-react/src/Auth/Provider/withGoogle.tsx]: @eknowles This PR is being replaced with #2436

**Comment by buggy** [comment]
@powerful23 I'm closing this and replacing it with #2436 which is now ready to go.

**Pull request closed without merging**